packages/loot-core/src/server/api-models.ts

```diff
@@ -153,3 +155,62 @@ export const budgetModel = {
     return file as Budget;
   },
 };
+
+export type AmountOPType = 'is' | 'isapprox' | 'isbetween';
+
+export type APIScheduleEntity = Pick<ScheduleEntity, 'id' | 'name'> & {
+  rule?: string; //All schedules has an associated underlying rule. not to be supplied iwth a new schedule
+  next_date?: string; //Next occurence of a schedule. not to be supplied iwth a new schedule
+  completed?: boolean; //not to be supplied iwth a new schedule
+  posts_transaction: boolean; //Whethere the schedule should auto post transaction on your behalf. Default to false
+  payee?: string | null; // Optional will default to null
+  account?: string | null; // Optional will default to null
+  amount: number | { num1: number; num2: number }; // Provide only 1 number except if the Amount
+  amountOp: AmountOPType; // 'is' | 'isapprox' | 'isbetween'
+  date: RecurConfig; // mandatory field in creating a schedule Mandatory field in creation
```

Comment:
the internal `RecurConfig` confuses me. It feels tough to try to represent a one-time schedule.

It would be great if `date` took just a single `Date` as well, e.g.:
```suggestion
  date: DateLike | RecurConfig;
```
(DateLike comes from [here](https://github.com/actualbudget/actual/blob/405c8b986f39dc7ef3e22295efc06163dca8b1bb/packages/loot-core/src/shared/months.ts#L10))